examples/accounts.py

```diff
@@ -0,0 +1,129 @@
+#deposit money into account
+parser_create = subparsers.add_parser('deposit', description = 'deposit money into account')
+parser_create.add_argument('account', help = 'account in which to deposit money')
+parser_create.add_argument('amount', help = 'total deposit', type = decimal.Decimal, default = decimal.Decimal('0.00'))
+parser_create.add_argument('memo', help = 'source of deposit', nargs = argparse.REMAINDER)
+#withdraw money from account (cash)
+parser_create = subparsers.add_parser('withdraw', description = 'withdraw amount of cash from account')
+parser_create.add_argument('account', help = 'account from which to withdraw money')
+parser_create.add_argument('amount', help = 'total withdrawl', type = decimal.Decimal, default = decimal.Decimal('0.00'))
+parser_create.add_argument('memo', help = 'reason for withdrawl', nargs = argparse.REMAINDER)
+args = parser.parse_args()
+
+#check to see if args.date is a string -- convert to datetime object if so
+if isinstance(args.date, str) == True:
+    args.date = datetime.datetime.strptime(args.date, '%Y-%m-%d')
+    
+with PersistentObjectPool(args.filename, flag='c') as pop:
+    if pop.root is None:
+        pop.root = pop.new(PersistentDict)
+    accounts = pop.root
+    if args.subcommand == 'create':
+        accounts[args.account] = [[args.date, decimal.Decimal(args.amount), 'Initial account balance']]
+        print("Created account '" + args.account + "'.")                       
+    elif args.subcommand == 'list':
+        L1 = ("Date         Amount  Balance  Memo\n"
+              "----------  -------  -------  ----")
+        print(L1)
+        accntBalance = decimal.Decimal(0)                           
+        for x in accounts[args.account]:
+            accntBalance = accntBalance + x[1]
+        for transaction in reversed(accounts[args.account]):
+            L2= "{:%Y-%m-%d}{:>9}{:>9}  {}".format(transaction[0], transaction[1], accntBalance, transaction[2])
+            print(L2)
+            accntBalance = accntBalance - transaction[1]
+    elif args.subcommand == 'transfer':
+        s= " "
+        memo = s.join(args.memo)
+        memo = memo[0].upper() + memo[1:]
+        accounts[args.pastLocation].append([args.date, -decimal.Decimal(args.transferAmount), memo])         
+        accounts[args.futureLocation].append([args.date, decimal.Decimal(args.transferAmount), memo])
+        pBalance = decimal.Decimal(0)
+        for transaction in accounts[args.pastLocation]:
+            pBalance = pBalance + transaction[1]
+        fBalance = decimal.Decimal(0)
+        for transaction in accounts[args.futureLocation]:
+            fBalance = fBalance + transaction[1]
+        print("Transferred {} from {} (new balance {}) to {} (new balance {})".format(args.transferAmount, args.pastLocation, str(pBalance), args.futureLocation, str(fBalance)))
+    elif args.subcommand == 'check':
+        c = " "
+        memo = c.join(args.memo)
+        memo = memo[0].upper() + memo[1:]
```

Comment:
You do this operation in several of the if cases.  That makes it a good candidate to turn into a function.  You'd use 'def format_memo(memo_parts):' up above the if and put the lines in the body of the function, and then in each case you can do 'memo = format_memo(args.memo)' instead of repeating your three (or two, see above) lines.